src/Beberlei/Metrics/Collector/InfluxDBOfficial.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Beberlei Metrics.
+ *
+ * LICENSE
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this package in the file LICENSE.txt.
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to kontakt@beberlei.de so I can send you a copy immediately.
+ */
+
+namespace Beberlei\Metrics\Collector;
+
+use InfluxDB\Database;
+use InfluxDB\Point;
+
+/**
+ * InfluxDB collector using official package influxdb/influxdb-php.
+ * This collector correctly sets timestamps to a point in time when the measurement was reported
+ * and not when it is flushed to influxdb server.
+ *
+ * Class InfluxDBOfficial
+ * @package Beberlei\Metrics\Collector
+ */
+class InfluxDBOfficial implements Collector, TaggableCollector
+{
+    /** @var \InfluxDB\Database */
+    private $database;
+
+    /** @var array */
+    private $data = array();
+
+    /** @var array */
+    private $tags = array();
+
+    /**
+     * @param Database $database
+     */
+    public function __construct(Database $database)
+    {
+        if (!class_exists('\InfluxDB\Database')) {
+            throw new \RuntimeException('Package \'influxdb/influxdb-php\' is required to use this collector.');
+        }
+
+        $this->database = $database;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function increment($variable)
+    {
+        $this->data[] = array($variable, 1, $this->getCurrentTimestamp());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function decrement($variable)
+    {
+        $this->data[] = array($variable, -1, $this->getCurrentTimestamp());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function timing($variable, $time)
+    {
+        $this->data[] = array($variable, $time, $this->getCurrentTimestamp());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function measure($variable, $value)
+    {
+        $this->data[] = array($variable, $value, $this->getCurrentTimestamp());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function flush()
+    {
+        foreach ($this->data as $data) {
+            $this->database->writePoints(
+                array(
+                    new Point(
+                        $data[0],
+                        $data[1],
+                        $this->tags,
+                        array(),
+                        $data[2]
+                    ),
+                ),
+                Database::PRECISION_SECONDS
+            );
+        }
+
+        $this->data = array();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setTags($tags)
+    {
+        $this->tags = $tags;
+    }
+
+    /**
+     * @return int
+     * @throws \Exception
```

Comment:
Why? The code is safe to me

Reply:
Well the constructor can throw an exception in case of an error.